src/tree/updater_gpu_hist.cu

```diff
@@ -47,8 +49,9 @@ struct GPUHistMakerTrainParam
                   "-1 to use all rows assignted to a GPU, and 0 to auto-deduce");
   }
 };
-
+#if !defined(GTEST_TEST)
```

Comment:
So we are excluding parameter registration when `GTEST_TEST` is defined? What benefit does it bring? Is this related to Google Test?

Reply:
@hcho3 No benefit.  Just to workaround the gpu hist test without overriding the file's git history.  See #4322